app/controllers/update_username_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require_dependency "#{Rails.root}/lib/importers/user_importer"
+
+class UpdateUsernameController < ApplicationController
+  before_action :require_signed_in
+  respond_to :html, :json
+  def index
+    @user = User
+  end
+
+  def update
+    @username = params['username']
```

Comment:
I think this would be a little clearer if this controller explicitly handled the sanitization, like this:

```ruby
username = UserImporter.sanitize_username params['username']
user = UserImporter.update_username_for_global_id username
```

That way, there will be no doubt that input sanitization is happening if someone reads this file but not UserImporter.

Also, I don't think the `username` variable needs to be an instance variable, as it's not relied upon by any other methods in this controller and it's not used by the render template.